### PR TITLE
Inherit annotations s3

### LINKED ISSUE #375: allow inheriting annotations from base trait/classes
* add `@inherit` annotation working on def's
* extend collecting annotations in magnolia based on that -> `annotationsOf(Symbol)`

### PR DESCRIPTION
**Pull request by kubinio123**

### CODE REVIEW

File: src/core/macro.scala
Comment (KacperFKorban) [inline src/core/macro.scala]: Comparing names is not 100% reliable, wouldn't comparing type symbols work?

File: src/core/macro.scala
Comment (KacperFKorban) [inline src/core/macro.scala]: Could you add an end marker or curly braces to this def?

File: src/core/macro.scala
Comment (KacperFKorban) [inline src/core/macro.scala]: Could you switch to curly braces here?

File: src/core/interface.scala
Comment (KacperFKorban) [inline src/core/interface.scala]: Does this import do anything anymore in Scala 3?

File: src/core/macro.scala
Comment (kubinio123) [reply src/core/macro.scala]: Thanks, seems to work when I just compare `s` against `tpe.typeSymbol`

File: src/core/interface.scala
Comment (kubinio123) [reply src/core/interface.scala]: I don't think so, not sure how it end up here :D

**Review by KacperFKorban — APPROVED** [review]
LGTM, but I would wait for @adamw approval as well :smile:

File: src/examples/show.scala
Comment (adamw) [inline src/examples/show.scala]: shouldn't we filter out these annotations from `inheritedAnns` in the first place?

File: src/core/magnolia.scala
Comment (adamw) [inline src/core/magnolia.scala:107]: can this impl call getParams_ to avoid code duplication?

File: src/core/interface.scala
Comment (adamw) [inline src/core/interface.scala:145]: same here, I'd just call param with the default inhertiedAnns value

File: src/core/macro.scala
Comment (adamw) [inline src/core/macro.scala:162]: maybe we should filter the base classes here as well?>

File: src/examples/show.scala
Comment (kubinio123) [reply src/examples/show.scala]: yep, after filtering out `Object` and `scala*` we don't need to filter these annotations

File: src/core/magnolia.scala
Comment (kubinio123) [reply src/core/magnolia.scala:107]: yes, missed that some how

**Comment by adamw** [comment]
Thanks @kubinio123 @KacperFKorban  :)

**Pull request merged**
